doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,72 @@
+
+## Versioning
+
+The official Azure CLI follows [Semantic Versioning](https://semver.org/) for version numbering. The CLI version consists of three parts `MAJOR.MINOR.PATCH`:
+- MAJOR version is upgraded only when there is a huge revolution. Generally in Azure CLI, this version is not upgraded.
+- MINOR version is upgraded for new features with the normal release process. All feature updates must be forward compatible, and breaking changes can only be published in the breaking change window *(Ignite sprint or Build sprint)*.
+- PATCH version is upgraded for hotfixes through the hotfix release process.
+
+## Life Cycle
+
+### Relase States
+
+Azure CLI differs from the [three life cycles](https://docs.microsoft.com/learn/modules/describe-service-life-cycle-microsoft-365/2-private-public-general-availability) of an Azure Service.  CLI only uses two states, **GA** (Generally Available) and **Preview**, making it easier to understand and maintain.
```

Comment:
We also have an `experimental` status used by both command groups, reference commands and parameters, correct?  

Can this information be added to the [Overview: Azure CLI terminology and support levels](https://learn.microsoft.com/en-us/cli/azure/reference-types-and-status?view=azure-cli-latest) article?  /or/ I can put in a "For more information see [this GitHub file](https://github.com/Azure/azure-cli/pull/24451/files)".

The target audience for this doc is (??) internal Microsoft service teams creating CLI source code?

Reply:
Good point, @zhoxing-ms, do we need to add experimental states for this? 

We should look at the layout of that and decide where this content should go. @dbradish-microsoft, feel free to set up a PR to get it started and we can discuss there. 

This audience is partners developing extensions and Deeply technical CLI users diving into the CLI to test and understand more.

Reply:
@dbradish-microsoft This is a good question! In fact, our team have discussed to deprecate the `is_experimental` flag. 
Because the role of `is_experimental` and `is_preview` are very similar, it is difficult for developers to distinguish which one should be used, and customters are often confused about their differences. 
Therefore, we plan to deprecate it, so as to make CLI release states become clearer, simpler and easier to understand

Reply:
@zhoxing-ms I am very much in favor of dropping the concept of an `experimental` status as I feel we should align with Azure having `public preview`, `preview` and `GA`.  `experimental` should become `public preview`, or we don't have a 3rd status at all.

@chasewilson, Regarding the target audience, developers creating extensions should expect documentation in GitHub.  Technical CLI users, however, I'd like to keep in MicrosoftDocs.

Reply:
@dbradish-microsoft Yes, the life states of CLI corresponds to the three states of Azure Service roughly as follows: *(this is roughly because the states of CLI commands are not always the same as api-version states)*
1. private preview: only build private packages, not publish them
2. public preview: add `is_preview` flag
3. GA: GA